core/src/main/kotlin/org/jetbrains/research/testspark/core/monitor/ErrorMonitor.kt

```diff
@@ -0,0 +1,19 @@
+package org.jetbrains.research.testspark.core.monitor
+
+interface ErrorMonitor {
+    var isErrorOccurred: Boolean
+
+    fun errorOccurred(): Boolean
+    fun clear() {
+        isErrorOccurred = false
+    }
+}
+
+class DefaultErrorMonitor : ErrorMonitor {
+    override var isErrorOccurred = false
+    override fun errorOccurred(): Boolean {
+        if (isErrorOccurred) return false
+        isErrorOccurred = true
+        return true
+    }
```

Comment:
The name of the method obscures its **rather tricky implementation**. I realized I did not understand how the `errorProcess` works until I revised this implementation.

Reply:
Generally, I tried to improve the namings of ErrorMonitor. It should be more clear now

Reply:
'+